src/Nethermind/Nethermind.BeaconNode.Peering/MothraPeeringWorker.cs

```diff
@@ -124,20 +125,43 @@ public override async Task StartAsync(CancellationToken cancellationToken)
 
                 _mothraLibp2p.Start(mothraSettings);
 
-                if (_logger.IsDebug()) LogDebug.PeeringWorkerStarted(_logger, null);
+                if (_logger.IsDebug()) LogDebug.PeeringWorkerExecuteCompleted(_logger, null);
             }
             catch (Exception ex)
             {
                 if (_logger.IsError()) Log.PeeringWorkerCriticalError(_logger, ex);
             }
+        }
 
-            await base.StartAsync(cancellationToken);
+        public override async Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (_logger.IsInfo())
+                Log.PeeringWorkerStarting(_logger, _clientVersion.Description,
+                    _environment.EnvironmentName, Thread.CurrentThread.ManagedThreadId, null);
+            await base.StartAsync(cancellationToken).ConfigureAwait(false);
         }
 
         public override async Task StopAsync(CancellationToken cancellationToken)
         {
             if (_logger.IsDebug()) LogDebug.PeeringWorkerStopping(_logger, null);
-            await base.StopAsync(cancellationToken);
+            await base.StopAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task EnsureInitializedWithAnchorState(CancellationToken stoppingToken)
+        {
+            // If the store is not initialized (no anchor block), then can not participate peer-to-peer
+            // e.g. can't send status message if we don't have a status.
+            // If this is pre-genesis, then the peering will wait until genesis has created the anchor block.
+            while (!_store.IsInitialized)
+            {
+                await Task.Delay(TimeSpan.FromMilliseconds(1000), stoppingToken).ConfigureAwait(false);
```

Comment:
shall there be a maximum wait time or this is absolutely required?

Reply:
This only applies if a node is launched pre-genesis (before actual genesis). Genesis is based on Eth1 blocks, with (a) minimum time + (b) minimum validators.

e.g. If we have min time 01 April, we would start the node running 31 March, but it has no genesis block (no one does), so there is no chain, and no point in peering with anyone; in fact, we can't send a status message, as we don't have a chain.

Some time after 01 April the genesis condition will be met, and all active nodes will simultaneously generate the same genesis block, then connect to each other, and start peering.

If a node joins later, in genesis mode, it needs to go back to the start of genesis day (April 01), and start scanning Eth1 until the genesis criteria are met, so it gets the same genesis block as everyone else.

However, once we know what the genesis block is, we can save it to disk, and release a new build that has the genesis block already encoded as the anchor block. This state would be read as part of startup, and no genesis check ever needed (the above code is not relevant).

Note: In fact, we may be behind other clients and never need genesis code.

Reply:
I'll add a debug message to output every minute while waiting for genesis to occur, but it does just have to wait.

Reply:
Message added